### PR TITLE
Use dedicated type for mass/weight values.  

### PR DESCRIPTION
**Pull request by BevapDin**
Basically the same as with `units::volume`: add and use a `units::mass` class as wrapper for values of weight.

Adds `_gram` and `_kilogram` literals and `to_kilogram` and `to_gram` functions (analog to `to_liter` and `to_milliliter`).

### CODE REVIEW

File: src/artifact.cpp [hunk 1]
```diff
@@ -1260,7 +1260,7 @@ void it_artifact_tool::serialize(JsonOut &json) const
     }
     json.end_array();
     json.member("volume", volume / units::legacy_volume_factor);
-    json.member("weight", weight);
+    json.member("weight", to_gram( weight ) );
```
Comment (AMurkin) [inline src/artifact.cpp]: missed namespace?

File: src/artifact.cpp [hunk 2]
```diff
@@ -1307,7 +1307,7 @@ void it_artifact_armor::serialize(JsonOut &json) const
...
```
Comment (AMurkin) [inline src/artifact.cpp]: missed namespace?

File: src/artifact.cpp (hunk 1 above)
Comment (BevapDin) [inline src/artifact.cpp]: No, it's the magic known as [ADL](http://en.cppreference.com/w/cpp/language/adl). The type of `weight` is defined in the `units` namespace, so `to_gram` is searched in that namespace as well.

Comment (AMurkin) [inline src/artifact.cpp]: Ok. Sorry.

**Comment by Coolthulhu** [comment]
Needs class/struct fixes - clang doesn't like when they're mixed up.

**Comment by kevingranade** [comment]
Also do make astyle, it's opt-out now so every new file must be astyle clean.

**Comment by codemime** [comment]
And also a rebase is now needed.

**Pull request merged**